.github/workflows/main.yml

```diff
@@ -29,9 +29,35 @@ jobs:
         with:
           ocaml-compiler: ${{ matrix.ocaml-compiler }}
           dune-cache: true
+          opam-pin: false
+          opam-depext: false
+
+      - run: opam pin . -y -n
 
       - run: opam install -t ambient-context ambient-context-lwt ambient-context-eio --deps-only
```

Comment:
Wait. Actually, instead of running all of these on one worker, with the same set of opam packages … part of the _intent_ of the support-matrix is to ensure consumers using `ambient-context` don't end up needing Lwt, or Melange, or etc — whatever bits they're not using;

so we should probably validate that with the tests? i.e. run each suite in a separate opam switch (or separate worker), _without the irrelevant dependencies installed at all_, because that's how our consumers will be consuming it?

Reply:
yeah the CI is a bit of a mess. I should probably split it into multiple files because this is too ugly.

Reply:
done. Simpler this way. No terrible yaml conditionals.